cpp/thirdparty/versions.txt

```diff
@@ -141,7 +141,7 @@ DEPENDENCIES=(
   "ARROW_AWS_CRT_CPP_URL aws-crt-cpp-${ARROW_AWS_CRT_CPP_BUILD_VERSION}.tar.gz https://github.com/awslabs/aws-crt-cpp/archive/${ARROW_AWS_CRT_CPP_BUILD_VERSION}.tar.gz"
   "ARROW_AWS_LC_URL aws-lc-${ARROW_AWS_LC_BUILD_VERSION}.tar.gz https://github.com/awslabs/aws-lc/archive/${ARROW_AWS_LC_BUILD_VERSION}.tar.gz"
   "ARROW_AWSSDK_URL aws-sdk-cpp-${ARROW_AWSSDK_BUILD_VERSION}.tar.gz https://github.com/aws/aws-sdk-cpp/archive/${ARROW_AWSSDK_BUILD_VERSION}.tar.gz"
-  "ARROW_BOOST_URL boost-${ARROW_BOOST_BUILD_VERSION}.tar.gz https://apache.jfrog.io/artifactory/arrow/thirdparty/7.0.0/boost_${ARROW_BOOST_BUILD_VERSION//./_}.tar.gz"
+  "ARROW_BOOST_URL boost-${ARROW_BOOST_BUILD_VERSION}.tar.gz https://archives.boost.io/release/1.86.0/source/boost_${ARROW_BOOST_BUILD_VERSION//./_}.tar.gz"
```

Comment:
We use a trimmed down version of boost (10mb vs 142mb) to speed up downloads.[ Could you create the new archive ](https://github.com/apache/arrow/blob/03c3f8ed41573b611c4c6dfaf1ec166a9b78decb/cpp/build-support/trim-boost.sh#L23)and upload it to the artifactory (as a committer you can do that) and use that version here?

Reply:
Will do. I just tried to run the script and am getting failures on the bootstrap/b2 step. I'll take a look later today.

Reply:
I didn't have any luck and it appears to be caused by Boost 1.86.0 releasing with a broken `bcp`. See comment and really the whole thread at https://github.com/boostorg/bcp/pull/18#issuecomment-2311880441.

I think we could probably update the `trim-boost` script to work around this but we could also just switch back to Thrift 0.20.0 and Boost 1.81.0 until Boost 1.87.0.